### PR TITLE
PLT-44 allow team switching without the need to login

### PR DESCRIPTION
**Pull request by coreyhulen**

### CODE REVIEW

File: api/user.go [hunk 1]
```diff
@@ -394,6 +394,41 @@ func Login(c *Context, w http.ResponseWriter, r *http.Request, user *model.User,
 
 	http.SetCookie(w, sessionCookie)
```
Comment (crspeller) [inline api/user.go]: Why are we still setting the original cookie up here?

File: web/react/components/navbar_dropdown.jsx [hunk 2]
```diff
@@ -142,10 +142,10 @@ export default class NavbarDropdown extends React.Component {
             >
             </li>
         );
-        if (this.state.teams.length > 1 && this.state.currentTeam) {
```
Comment (crspeller) [inline web/react/components/navbar_dropdown.jsx]: Removing this.state.currentTeam here causes the line `, currentTeam: TeamStore.getCurrent()` in the constructor to be irrelevant. Should be removed.

File: web/web.go [hunk 3]
```diff
@@ -297,6 +332,19 @@ func getChannel(c *api.Context, w http.ResponseWriter, r *http.Request) {
...
 
+	if tResult := <-teamChan; tResult.Err != nil {
+		c.Err = tResult.Err
+		return
+	} else {
+		team = tResult.Data.(*model.Team)
+	}
+
+	if team.Name != teamName {
+		l4g.Error("It appears you are log into " + team.Name + ", but are trying to access " + teamName)
```
Comment (crspeller) [inline web/web.go]: Is this really a server error?

Comment (jwilander) [inline web/web.go]: Also spelling error, should be ".. logged into.." instead of ".. log into.."

File: api/user.go [hunk 4]
```diff
@@ -394,6 +394,41 @@ func Login(c *Context, w http.ResponseWriter, r *http.Request, user *model.User,
...
+	if len(multiToken) > 0 {
+		tokens := strings.Split(multiToken, " ")
+
+		multiToken = ""
+		seen := make(map[string]string)
+		seen[session.TeamId] = session.TeamId
+		for _, token := range tokens {
+			if sr := <-Srv.Store.Session().Get(token); sr.Err == nil {
+				s := sr.Data.(*model.Session)
+				if !s.IsExpired() && seen[s.TeamId] == "" {
```
Comment (jwilander) [inline api/user.go]: Minor, 0/5 on whether it's changed but go convention to check for key existence is to do this:

```
if _, ok := seen[s.TeamId]; !s.IsExpired() && !ok {
    //stuff
}
```

File: web/web.go [hunk 5]
```diff
@@ -297,6 +332,19 @@ func getChannel(c *api.Context, w http.ResponseWriter, r *http.Request) {
...
+		http.Redirect(w, r, c.GetSiteURL()+"/"+team.Name+"/channels/town-square", http.StatusFound)
```
Comment (jwilander) [inline web/web.go]: Also minor, can we use `c.GetTeamURL()` or 'c.GetTeamURLFromTeam()' here?

File: api/user.go (hunk 1 above)
Comment (coreyhulen) [inline api/user.go]: We still use the original cookie.  I've added a second to manage the different teams.  MMSID is your current team and MMSIDMU is the list of possible teams.

File: web/react/components/navbar_dropdown.jsx (hunk 2 above)
Comment (coreyhulen) [inline web/react/components/navbar_dropdown.jsx]: will do

File: web/web.go (hunk 3 above)
Comment (coreyhulen) [inline web/web.go]: It's an error, but I force a redirect to the correct team.

**Comment by coreyhulen** [comment]
@jwilander @crspeller issues fixed.  ready for re-review

**Comment by coreyhulen** [comment]
Wait the redirect issue was intentional.  Basically it cannot trust the url it needs to get it via the session which is different.

**Comment by jwilander** [comment]
+1

**Comment by crspeller** [comment]
+1

**Pull request merged**